### DISCUSSION TITLE
Search results with Eleventy Edge

### DISCUSSION THREAD

**Discussion by mark-reason** [11ty/eleventy#2344] [Uncategorized]
Hi,

I'm looking to fetch search results via Eleventy Edge. So I will hit an endpoint, it will respond with some JSON, I will then loop through these and return HTML. Is this something Edge would be able to handle or am I barking up the wrong tree?

Cheers

**Comment by cfjedimaster** [comment]
Edge won't do search for you, you would need to handle that yourself somehow, and it can also be done with Eleventy Serverless, regular ole Netlify Functions, or heck, client side. You've got a lot of options. Some examples:

https://www.raymondcamden.com/2021/06/06/using-lunr-with-eleventy-via-netlify-serverless-functions-part-two
https://www.raymondcamden.com/2021/06/02/using-lunr-with-eleventy-via-netlify-serverless-functions
https://www.raymondcamden.com/2021/01/22/using-pre-built-lunr-indexes-with-eleventy
https://www.raymondcamden.com/2020/07/01/adding-algolia-search-to-eleventy-and-netlify-part-two
https://www.raymondcamden.com/2020/06/24/adding-algolia-search-to-eleventy-and-netlify

**Comment by p3palazzo** [comment]
I've been trying out these solutions with client-side Lunr search on Eleventy 2.0.1. So far I have in my base layout (all the logs are there for debugging):

```
<head>
<script src="/assets/lunr/lunr.min.js"></script>
<script>
let idx;

fetch('/lunr.json')
  .then(response => {
    if (!response.ok) {
      throw new Error(`HTTP error! status: ${response.status}`);
    }
    return response.json();
  })
  .then(data => {
    console.log("Index data loaded:", data); // Log data
    idx = lunr.Index.load(data);
    console.log("Lunr index created:", idx); // Log index creation
  })
  .catch(error => {
    console.error("Error loading lunr.json:", error); // Log errors
  });

function performSearch(event) {
  event.preventDefault();
  const query = document.getElementById('search').value.trim();
  console.log("Search query:", query); // Log search query

  if (idx) {
    const results = idx.search(query);
    console.log("Search results:", results); // Log search results

    localStorage.setItem('searchResults', JSON.stringify(results));
    window.location.href = '/search/index.html';
  } else {
    console.error("Lunr index not initialized");
  }
  return false;
}
</script>
</head>
```

Now when I load my search page, I'd like to pass the search results to the `pagination.items` variable in the template. Is this possible at all? My search page has this:

```
<script>
function loadSearchResults() {
  const results = JSON.parse(localStorage.getItem('searchResults'));
  if (results) {
    return results;
  } else {
    console.error("No search results found in localStorage");
    return [];
  }
}

searchResults = loadSearchResults();
console.log("global results:", searchResults);
</script>
```

For which I am getting searchResults as `[ { ref: "n", score: "n", matchData: {...} }, { ... }, ... ]`. The results all belong to a single collection.

I also looked at [this issue](https://github.com/mozilla/nunjucks/issues/918) but it seems I would need to render the template from within the script, and I'm not sure how to do this.

**Comment by rdela** [comment]
@p3palazzo did you evaluate [Pagefind](https://pagefind.app/)? 

Some resources: 

- [Using PageFind with Eleventy for Search](https://rknight.me/blog/using-pagefind-with-eleventy-for-search/)
- [Adding pagefind to my Eleventy personal site](https://www.bobmonsour.com/posts/adding-pagefind-to-my-eleventy-personal-site.md/)
- [CloudCannon/alto-eleventy-template](https://github.com/CloudCannon/alto-eleventy-template)

**Reply by p3palazzo** [reply]
I'll give it a try, thanks for the heads up!

**Reply by p3palazzo** [reply]
Update: Thanks a lot for the recommendation, pageFind works very well out of the box and I’m now trying to understand the API in order to build custom results pages.

**Reply by rdela** [reply]
@p3palazzo nice the 11ty website implementation https://www.11ty.dev/docs/search/ - source https://github.com/11ty/11ty-website/blob/main/src/docs/search.njk + https://github.com/11ty/11ty-website/blob/main/src/_includes/components/search.js may help.
